### ISSUE TITLE
Promisified queries are returned as a MongoServer error after 30 seconds

### ISSUE THREAD

**Issue by tomgrossman** [Automattic/mongoose#4248]
I have reverted back to version 4.1.14 to avoid this issue.
So it looks like (i think) it's related to the feature
**fix(connection): emit on error if listeners and no callback #4098**
from version 4.4.15.

I am using Bluebirds **promisifyAll** on mongoose, and I used an aggregateAsync query on the DB.
this query should take about 40-50 seconds (when I tested it manually), but mongoose returned an error after 30 seconds all the time: 

`MongoError: connection 1 to xxx:27017 timed out`

**Comment by vkarpov15** [comment]
Can you provide a more substantial code sample? Also can you try using `mongoose.Promise = require('bluebird')` instead of promisify?

**Comment by tomgrossman** [comment]
mm I just ran an `aggregateAsync()` query to group a large collection.
But for some reason I can't reproduce it now, so I guess you can close it.

**Comment by erikrenberg** [comment]
I got the exact same timeout with the latest 4.6.4 mongoose release:

The cause for me was a large(ish) database for which the indexes in a large collection haven't been been built yet when my mongoose app is connecting to the database and doing a find() which then times out after 30 seconds.

Once the indexes have been completely built (as can be seen in the MongoDB logs) mongoose no longer times out. This might explain why @tomgrossman couldn't repro in his case.

Anyway; to fix, increase the `socketTimeoutMS` to something larger, for example 300000 (5 minutes) when calling mongoose.connect():

```
mongoose.connect(config.mongo.uri,  { 
  server: { 
    socketOptions: { socketTimeoutMS: 300000 }
  } } 
);
```

The 30 second default time out comes from the Node MongoDB driver used: http://mongodb.github.io/node-mongodb-native/2.2/reference/connecting/connection-settings/

The fix mentioned above for 4.4.15 (https://github.com/Automattic/mongoose/commit/df56f2f4c836a2f42fdff70e81b7db12057535ff) simply surfaced this timeout. Before it probably happened as well but wasn't reported because `_this.error(err, callback)` wasn't called when the database was opened successfully beginning on line 493.
